app/mailers/bill_mailer.rb

```diff
@@ -0,0 +1,57 @@
+# typed: true
+class BillMailer < ApplicationMailer
+  extend T::Sig
+
+  sig do
+    params(
+        to_name: String,
+        to_email: String,
+        from: String,
+        for_message: String,
+        count: Integer,
+        from_email: T.nilable(String)
+      )
+      .returns(Mail::Message)
+  end
+  def bill(to_name:, to_email:, from:, for_message:, count: 1, from_email: nil)
+    @bucks =
+      bucks(
+        to_name: to_name,
+        from: from,
+        for_message: for_message,
+        count: count
+      )
+    @bucks.each do |file_name, buck|
+      attachments.inline[file_name] = buck.to_blob
+    end
+    mail(to: to_email, subject: "You've got Bills (Bucks)")
+  end
+
+  private
+
+  sig do
+    params(to_name: String, from: String, for_message: String, count: Integer)
+      .returns(T::Hash[String, Buck])
+  end
+  def bucks(to_name:, from:, for_message:, count: 1)
```

Comment:
Why do this in the mailer? I thought we already had the model/controller creating bucks?

Reply:
Good point, I thought about making a BuckCollection class that can be passed from the controller to the mailer. I'll give that a try.